packages/utils/src/calls.rs

```diff
@@ -0,0 +1,185 @@
+use serde::{de::DeserializeOwned, Serialize};
+
+use cosmwasm_std::{
+    to_binary, Coin, CosmosMsg, HumanAddr, Querier, QueryRequest, StdResult, Uint128, WasmMsg,
+    WasmQuery,
+};
+
+use super::space_pad;
+
+/// A data structure representing the instantiation message of a contract
+///
+/// This trait provides a method to create the CosmosMsg used to instantiate a contract
+pub trait InitCallback: Serialize {
+    /// pad the message to blocks of this size
+    const BLOCK_SIZE: usize;
+
+    /// Returns StdResult<CosmosMsg> used to instantiate the specified contract
+    ///
+    /// # Arguments
+    ///
+    /// * `label` - String holding the label for the new contract instance
+    /// * `code_id` - code ID of the contract to be instantiated
+    /// * `callback_code_hash` - String holding the code hash of the contract to be instantiated
+    /// * `send_amount` - Optional Uint128 amount of native coin to send with instantiation message
```

Comment:
This docstring is good, but I'd change the main description. I'd write something like
>Tries to convert the instance of the implementing type to a `CosmosMsg`. The `BLOCK_SIZE` specified in the implementation is used when padding the message

but like in your version, mention that it's for init messages specifically. Maybe something like "a `CosmosMsg` that can will trigger instantiation of a contract"